### ISSUE TITLE
chore(core): rename Uni_IsSMP to Uni_IsNonBMP

### ISSUE THREAD

**Issue by srl295** [keymanapp/keyman#10654]
Suggest to rename `Uni_IsSMP()` to `Uni_IsNonBMP()` across unicode.h and kmcompx.h as it is misleading. The SMP is exactly U+10000..U+1FFFF only. Only used in a few places.

Discussed this at #10557.

**Comment by mcdurdin** [comment]
Yes my bad, back in probably 1998. SMP stands for Some More Planes and covers everything that is not in the BMP. Doesn't it?

**Comment by mcdurdin** [comment]
To be honest, the name is still ambiguous: what data type does it accept? I guess it takes a USV (u)int32_t but... that's different to some of the other `Uni_` macros. 

The `Uni_` class of macros in unicode.h could all be reviewed for better names.
